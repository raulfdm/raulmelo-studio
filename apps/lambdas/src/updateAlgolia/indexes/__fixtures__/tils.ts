export const tilsApiResponse = {
  data: {
    tils: [
      {
        id: '6081113c532c3c28f38780df',
        publishedAt: '2021-04-19',
        title: 'How to format Rust code "on save" in VSCODE',
        content:
          'For some reason, when we install Rust VSCODE extension the formatter does not work by default.\nTo do that I we need to enforce the formatter via `settings.json` (or User preference)\n\n```rust\n{\n    "[rust]": {\n        "editor.defaultFormatter": "rust-lang.rust", // Makes the magic\n        "editor.formatOnSave": true // Optional\n    },\n}\n``',
        locale: 'en',
        slug: 'how-to-format-rust-code-on-save-in-vscode',
      },
    ],
  },
};
