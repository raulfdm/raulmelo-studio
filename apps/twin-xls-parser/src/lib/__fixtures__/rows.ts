import type { Row } from 'read-excel-file';

export const rowsMock = [
	[
		null,
		'ASSESSORIA ESPORTIVA TWIN',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		'TREINO DE HIPERTROFIA',
		null
	],
	[
		null,
		'Treino A',
		null,
		null,
		null,
		'Treino B',
		null,
		null,
		null,
		'Treino C',
		null,
		null,
		null,
		'Musculação',
		null
	],
	[
		null,
		'Dorsais + Antebraço',
		null,
		null,
		null,
		'Peitoral + Abdômen',
		null,
		null,
		null,
		'Deltóides + Trapézio',
		null,
		null,
		null,
		'Seg -',
		'A'
	],
	[
		null,
		'Exercícios',
		'SxR',
		'Técnica Avançada',
		null,
		'Exercícios',
		'SxR',
		'Técnica Avançada',
		null,
		'Exercícios',
		'SxR',
		'Técnica Avançada',
		null,
		'Ter -',
		null
	],
	[
		null,
		'Levantamento Terra',
		'4x12',
		null,
		null,
		'Supino Reto Barra',
		'5x12',
		"Rest 'n' Pause 3x",
		null,
		'Desenvolvimento Halteres',
		'4x8',
		null,
		null,
		'Qua -',
		'B'
	],
	[
		null,
		'Puxada Vertical Peg Supinada + Puxada Vertical Peg Pronada',
		'4x10',
		'Bi-Set',
		null,
		'Crucifixo Reto Halteres',
		'5x10',
		null,
		null,
		'Elevação Lateral Halteres',
		'4x8',
		"Rest 'n' Pause 3x",
		null,
		'Qui -',
		null
	],
	[
		null,
		'Remada Unilateral Peg Supinada',
		'4x12',
		null,
		null,
		'Peck Deck ou Crucifixo Na Máquina',
		'5x8',
		'Drop-Set 3x',
		null,
		'Elevação Frontal Halteres Peg Pronada',
		'4x8',
		null,
		null,
		'Sex -',
		'C'
	],
	[
		null,
		'Flexão de Punho Barra Sentado',
		'3x12',
		null,
		null,
		'Supra',
		'4x10',
		'Descanso 30s',
		null,
		'Remada Alta Smith',
		'5x8',
		null,
		null,
		'Sáb -',
		'D'
	],
	[
		null,
		'Extensão de Punho Barra Sentado',
		'2x8',
		null,
		null,
		'Oblíquos',
		'4x15',
		'Descanso 30s',
		null,
		'Encolhimento Barra Anterior + Remada Alta Halteres',
		'5x8',
		'Bi-Set',
		null,
		'Dom -',
		'E'
	],
	[
		null,
		'Rosca Inversa Cross',
		'3x10',
		'Drop-Set 3x',
		null,
		'Prancha',
		'4xF',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, 'Aeróbicos', null],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, 'Seg -', '10min'],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, 'Ter -', '40min'],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, 'Qua -', '10min'],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, 'Qui -', '40min'],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, 'Sex -', '10min'],
	[
		null,
		'INTERVALO ENTRE SÉRIES E EXERCÍCIOS: 45 a 60s',
		null,
		null,
		null,
		'INTERVALO ENTRE SÉRIES E EXERCÍCIOS: 45 a 60s',
		null,
		null,
		null,
		'INTERVALO ENTRE SÉRIES E EXERCÍCIOS: 45 a 60s',
		null,
		null,
		null,
		'Sáb -',
		null
	],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, 'Dom -', '10min'],
	[
		null,
		'Treino D',
		null,
		null,
		null,
		'Treino E',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Panturrilha + Coxa',
		null,
		null,
		null,
		'Bíceps + Tríceps + Abdômen',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Exercícios',
		'SxR',
		'Técnica Avançada',
		null,
		'Exercícios',
		'SxR',
		'Técnica Avançada',
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Gêmeos Sentado Livre',
		'4x12',
		null,
		null,
		'Rosca Alternada Cross-Over',
		'4x12',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Gêmeos Em Pé Máquina',
		'4x12',
		null,
		null,
		'Rosca Direta Halteres',
		'4x10',
		'Drop-Set 3x',
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Gêmeos Leg Press',
		'2x10',
		null,
		null,
		'Rosca Concentrada',
		'4x8',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Cadeira Flexora',
		'5x12',
		"Rest 'n' Pause 3x",
		null,
		'Tríceps Pulley Corda',
		'4x12',
		'Drop-Set 3x',
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Cadeira Adutora',
		'5x10',
		null,
		null,
		'Tríceps Pulley Unilateral Peg Pronada',
		'4x10',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Agachamento Livre',
		'5x8',
		null,
		null,
		'Tríceps Pulley Unilateral Peg Supinada',
		'4x8',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Cadeira Abdutora',
		'4x10',
		null,
		null,
		'Oblíquos',
		'4x12',
		'Descanso 30s',
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Glúteos Graviton',
		'4x12',
		null,
		null,
		'Infra',
		'4x10',
		'Descanso 30s',
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[
		null,
		'Glúteos Máquina',
		'4x10',
		null,
		null,
		'Prancha',
		'4xF',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, null, null],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, null, null],
	[null, null, null, null, null, null, null, null, null, null, null, null, null, null, null],
	[
		null,
		'INTERVALO ENTRE SÉRIES E EXERCÍCIOS: 60 a 90s',
		null,
		null,
		null,
		'INTERVALO ENTRE SÉRIES E EXERCÍCIOS: 45 a 60s',
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null,
		null
	]
] as Row[];
