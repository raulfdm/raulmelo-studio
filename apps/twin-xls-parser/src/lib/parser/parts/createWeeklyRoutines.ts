import dayjs from 'dayjs';

import { Routine } from '../../models/Routine';

export function createWeeklyRoutines(startDate: string, endDate: string) {
	const startDay = dayjs(startDate);
	const endDay = dayjs(endDate);

	const diff = endDay.diff(startDay, 'day') + 1; // +1 because we want to include the end date

	const result: Routine[] = [];

	for (let currDay = 0; currDay < diff; currDay++) {
		result.push(new Routine(startDay.add(currDay, 'day').format('YYYY-MM-DD')));
	}

	return result;
}
