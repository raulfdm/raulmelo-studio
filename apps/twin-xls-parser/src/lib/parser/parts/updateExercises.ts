import type { Row } from 'read-excel-file';

import { Exercise } from '../../models/Exercise';
import type { Routine } from '../../models/Routine';

export function updateExercises(rows: Row[], weekRoutine: Routine[]): void {
	const trainingRoutine = getOnlyTraining(weekRoutine);

	/**
	 * This values are based on the excel file structure.
	 *
	 * The first block over exercises starts at row 1 and ends at row 17.
	 * The second block over exercises starts at row 18 and ends at the end of the file.
	 */
	updateExercises(1, 17, [0, 1, 2]);
	updateExercises(18, undefined, [3, 4, 5]);

	function updateExercises(rowStart: number, rowEnd: number, trainings: [number, number, number]) {
		const nextRows = rows.slice(rowStart, rowEnd);
		const subtitleRowIndex = 1;
		const exerciseRowIndexStart = 3;
		const exerciseRowIndexEnd = 14;
		const suggestedRestRowIndex = 15;

		for (let rowIndex = 0; rowIndex < nextRows.length; rowIndex++) {
			const row = nextRows[rowIndex];

			if (rowIndex === subtitleRowIndex) {
				handleSubtitle(row);
			} else if (rowIndex >= exerciseRowIndexStart && rowIndex <= exerciseRowIndexEnd) {
				handleExercises(row);
			} else if (rowIndex === suggestedRestRowIndex) {
				handleSuggestedRest(row);
			}
		}

		function handleSubtitle(row: Row) {
			const columnIndexes = [1, 5, 9];

			for (let index = 0; index < columnIndexes.length; index++) {
				const columnIndex = columnIndexes[index];
				const routineIndex = trainings[index];

				const cellValue = row[columnIndex];

				const currentRoutine = trainingRoutine[routineIndex];

				if (currentRoutine && cellValue) {
					currentRoutine.subtitle = cellValue as string;
				}
			}
		}

		function handleExercises(row: Row) {
			const columnsIndexes = [
				[1, 2, 3],
				[5, 6, 7],
				[9, 10, 11]
			];

			for (let index = 0; index < columnsIndexes.length; index++) {
				const exerciseIndexes = columnsIndexes[index];
				const routineIndex = trainings[index];
				const routine = trainingRoutine[routineIndex];

				const name = row[exerciseIndexes[0]] as string;
				const sXR = row[exerciseIndexes[1]] as string;
				const advancedTechnique = row[exerciseIndexes[2]] as string;

				if (name !== null && sXR !== null) {
					// Bi-set
					if (name.includes('+')) {
						const [ex1, ex2] = name.split('+');

						const exercise1 = new Exercise();
						exercise1.name = ex1.trim();

						exercise1.setSxR = sXR;
						exercise1.advancedTechnique = advancedTechnique;
						routine.addExercise(exercise1);

						const exercise2 = new Exercise();
						exercise2.name = ex2.trim();
						exercise2.setSxR = sXR;
						exercise2.advancedTechnique = advancedTechnique;
						routine.addExercise(exercise2);
					} else {
						const exercise = new Exercise();
						exercise.name = name;
						exercise.setSxR = sXR;
						exercise.advancedTechnique = advancedTechnique;
						routine.addExercise(exercise);
					}
				}
			}
		}

		function handleSuggestedRest(row: Row) {
			const columnIndexes = [1, 5, 9];

			for (let index = 0; index < columnIndexes.length; index++) {
				const columnIndex = columnIndexes[index];
				const routineIndex = trainings[index];

				const cellValue = row[columnIndex];

				const currentRoutine = trainingRoutine[routineIndex];

				if (cellValue !== null) {
					currentRoutine.suggestedRest = cellValue as string;
				}
			}
		}
	}
}

function getOnlyTraining(routines: Routine[]) {
	return routines.filter((routine) => routine.title.includes('Treino'));
}
