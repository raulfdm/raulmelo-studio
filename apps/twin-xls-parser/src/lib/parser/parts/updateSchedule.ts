import type { Row } from 'read-excel-file';

import type { Routine } from '../../models/Routine';

export function updateSchedule(rows: Row[], weekRoutine: Routine[]) {
	const rowStart = 1;
	const rowEnd = 9;

	const columnEnd = 14;
	let dayIndex = 0;

	for (let rowIndex = 0; rowIndex < rows.length; rowIndex++) {
		if (rowIndex > rowStart && rowIndex < rowEnd) {
			const row = rows[rowIndex];

			for (let columnIndex = 0; columnIndex < row.length; columnIndex++) {
				if (columnIndex == columnEnd) {
					const cellValue = row[columnIndex];

					const currentRoutine = weekRoutine[dayIndex];

					if (cellValue === null) {
						currentRoutine.title = 'Cardio';
						currentRoutine.subtitle = 'AEJ';
					} else {
						currentRoutine.title = `Treino ${cellValue}`;
					}

					dayIndex++;
				}
			}
		}
	}
}
