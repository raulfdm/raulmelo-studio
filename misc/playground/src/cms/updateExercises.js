import sanityClient from '@sanity/client';
import { config } from 'dotenv';
import got from 'got';
import groq from 'groq';

config();

const raulClient = sanityClient({
  projectId: 'gc3hakk3',
  dataset: 'production',
  apiVersion: 'v1',
  useCdn: false,
});

const camilaClient = sanityClient({
  projectId: '17miagfv',
  dataset: 'production',
  apiVersion: 'v1',
  useCdn: false,
  token: process.env.CAMILA_SANITY_TOKEN,
});

const query = groq`
*[_type == "exercise"]{
  ...,
  "image": image.asset-> {
    url
  }
}
`;
let counter = 0;
const fails = [];

try {
  const exercises = await raulClient.fetch(query);
  console.log('Number of exercises:', exercises.length);

  exercises.forEach((exercise) => {
    if (exercise.image === undefined) {
      console.log(exercise._id);
    }
  });

  const transactions = camilaClient.transaction();

  console.log('Start the migration');
  for await (const exercise of exercises) {
    console.group();
    const { _id, _type, name, video, image } = exercise;

    console.log(`${++counter}. ${name}`);

    try {
      const imageStream = got.stream(image.url);
      console.log('Uploading image');
      const sanityImage = await camilaClient.assets.upload(
        'image',
        imageStream,
      );
      console.log('Image uploaded');

      transactions.createOrReplace({
        _id,
        _type,
        name,
        video,
        image: {
          _type: 'image',
          asset: {
            _ref: sanityImage._id,
            _type: 'reference',
          },
        },
      });
      console.log('Item added\n--------');
    } catch (error) {
      console.log('FAIL');
      fails.push({ _id, error, exercise });
    }

    console.groupEnd();
  }

  console.log('Committing the transaction');
  await transactions.commit();
  console.log('Migration completed');
} catch (error) {
  console.log(error);
}

if (fails.length > 0) {
  console.log(fails);
}
