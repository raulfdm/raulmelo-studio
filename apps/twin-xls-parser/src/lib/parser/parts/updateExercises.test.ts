import { rowsMock } from '../../__fixtures__/rows';
import { createWeeklyRoutines } from './createWeeklyRoutines';
import { updateExercises } from './updateExercises';
import { updateSchedule } from './updateSchedule';

describe('fn: updateExercises', () => {
	it('updates all exercise correctly', () => {
		const weekRoutine = createWeeklyRoutines();
		updateSchedule(rowsMock, weekRoutine);

		updateExercises(rowsMock, weekRoutine);

		expect(weekRoutine.map((r) => r.getObject())).toMatchSnapshot();
	});
});
