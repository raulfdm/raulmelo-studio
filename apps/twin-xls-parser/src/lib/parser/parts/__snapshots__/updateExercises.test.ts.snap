// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`fn: updateExercises > updates all exercise correctly 1`] = `
[
  {
    "cardioTime": 0,
    "date": "2023-01-08",
    "exercises": [
      {
        "advancedTechnique": null,
        "name": "Levantamento Terra",
        "reps": "12",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": undefined,
        "name": "Puxada Vertical Peg Supinada",
        "reps": "10",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": undefined,
        "name": "Puxada Vertical Peg Pronada",
        "reps": "10",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Remada Unilateral Peg Supinada",
        "reps": "12",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Flexão de Punho Barra Sentado",
        "reps": "12",
        "rest": 45,
        "series": 3,
      },
      {
        "advancedTechnique": null,
        "name": "Extensão de Punho Barra Sentado",
        "reps": "8",
        "rest": 45,
        "series": 2,
      },
      {
        "advancedTechnique": undefined,
        "name": "Rosca Inversa Cross",
        "reps": "10",
        "rest": 45,
        "series": 3,
      },
    ],
    "id": "8e162e4cf8dd2898718c5e2cebbdfa29c8b5d01624a842ed71f727183cdf5224",
    "subtitle": "Dorsais + Antebraço",
    "title": "Treino A",
  },
  {
    "cardioTime": 0,
    "date": "2023-01-09",
    "exercises": null,
    "id": "b8d54423b3ce6cfc3329c7054348a0eb3d544a8bb57fffa8a939816e372ed472",
    "subtitle": "AEJ",
    "title": "Cardio",
  },
  {
    "cardioTime": 0,
    "date": "2023-01-10",
    "exercises": [
      {
        "advancedTechnique": "rest_and_pause",
        "name": "Supino Reto Barra",
        "reps": "12",
        "rest": 45,
        "series": 5,
      },
      {
        "advancedTechnique": null,
        "name": "Crucifixo Reto Halteres",
        "reps": "10",
        "rest": 45,
        "series": 5,
      },
      {
        "advancedTechnique": undefined,
        "name": "Peck Deck ou Crucifixo Na Máquina",
        "reps": "8",
        "rest": 45,
        "series": 5,
      },
      {
        "advancedTechnique": null,
        "name": "Supra",
        "reps": "10",
        "rest": 30,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Oblíquos",
        "reps": "15",
        "rest": 30,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Prancha",
        "reps": "F",
        "rest": 45,
        "series": 4,
      },
    ],
    "id": "9586d4448a832f8179a226fc6ffcd32b2a24cc6a04ff31e84c0e25e0acd52f5c",
    "subtitle": "Peitoral + Abdômen",
    "title": "Treino B",
  },
  {
    "cardioTime": 0,
    "date": "2023-01-11",
    "exercises": null,
    "id": "23eded54599c71f8c83599eafb8c6f1ff857053dbe7fd1df8f6cd84312956f57",
    "subtitle": "AEJ",
    "title": "Cardio",
  },
  {
    "cardioTime": 0,
    "date": "2023-01-12",
    "exercises": [
      {
        "advancedTechnique": null,
        "name": "Desenvolvimento Halteres",
        "reps": "8",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": "rest_and_pause",
        "name": "Elevação Lateral Halteres",
        "reps": "8",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Elevação Frontal Halteres Peg Pronada",
        "reps": "8",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Remada Alta Smith",
        "reps": "8",
        "rest": 45,
        "series": 5,
      },
      {
        "advancedTechnique": undefined,
        "name": "Encolhimento Barra Anterior",
        "reps": "8",
        "rest": 45,
        "series": 5,
      },
      {
        "advancedTechnique": undefined,
        "name": "Remada Alta Halteres",
        "reps": "8",
        "rest": 45,
        "series": 5,
      },
    ],
    "id": "6944197def090466ac43ba3c84d816d537c3e180858486cd541460c002305388",
    "subtitle": "Deltóides + Trapézio",
    "title": "Treino C",
  },
  {
    "cardioTime": 0,
    "date": "2023-01-13",
    "exercises": [
      {
        "advancedTechnique": null,
        "name": "Gêmeos Sentado Livre",
        "reps": "12",
        "rest": 60,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Gêmeos Em Pé Máquina",
        "reps": "12",
        "rest": 60,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Gêmeos Leg Press",
        "reps": "10",
        "rest": 60,
        "series": 2,
      },
      {
        "advancedTechnique": "rest_and_pause",
        "name": "Cadeira Flexora",
        "reps": "12",
        "rest": 60,
        "series": 5,
      },
      {
        "advancedTechnique": null,
        "name": "Cadeira Adutora",
        "reps": "10",
        "rest": 60,
        "series": 5,
      },
      {
        "advancedTechnique": null,
        "name": "Agachamento Livre",
        "reps": "8",
        "rest": 60,
        "series": 5,
      },
      {
        "advancedTechnique": null,
        "name": "Cadeira Abdutora",
        "reps": "10",
        "rest": 60,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Glúteos Graviton",
        "reps": "12",
        "rest": 60,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Glúteos Máquina",
        "reps": "10",
        "rest": 60,
        "series": 4,
      },
    ],
    "id": "abc66f68b12371a7a8f3cfefc2465f8383cb98285dcddb9f169754b73272ec8c",
    "subtitle": "Panturrilha + Coxa",
    "title": "Treino D",
  },
  {
    "cardioTime": 0,
    "date": "2023-01-14",
    "exercises": [
      {
        "advancedTechnique": null,
        "name": "Rosca Alternada Cross-Over",
        "reps": "12",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": undefined,
        "name": "Rosca Direta Halteres",
        "reps": "10",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Rosca Concentrada",
        "reps": "8",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": undefined,
        "name": "Tríceps Pulley Corda",
        "reps": "12",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Tríceps Pulley Unilateral Peg Pronada",
        "reps": "10",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Tríceps Pulley Unilateral Peg Supinada",
        "reps": "8",
        "rest": 45,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Oblíquos",
        "reps": "12",
        "rest": 30,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Infra",
        "reps": "10",
        "rest": 30,
        "series": 4,
      },
      {
        "advancedTechnique": null,
        "name": "Prancha",
        "reps": "F",
        "rest": 45,
        "series": 4,
      },
    ],
    "id": "3c943f48f379151df2446bc3d0cc821be49b5296355923357719e380e3266f88",
    "subtitle": "Bíceps + Tríceps + Abdômen",
    "title": "Treino E",
  },
]
`;
